src/app/scenario/scenario.component.ts

```diff
@@ -65,7 +65,7 @@ export class ScenarioComponent {
         imports ? imports : []
       ],
       declarations: [
-        type,
+        ...(sandbox.declareComponent ? [ type ] : []),
```

Comment:
Nice! We should probably change the `declarations ? declarations : []` to use the spread operator too. Can do that in a separate commit.